pilot/pkg/model/push_context.go

```diff
@@ -870,6 +885,28 @@ func (ps *PushContext) DestinationRule(proxy *Proxy, service *Service) *config.C
 	return nil
 }
 
+func (ps *PushContext) DestinationRuleByName(proxy *Proxy, name, namespace string) *config.Config {
+	if proxy == nil || proxy.SidecarScope == nil {
+		return nil
+	}
+	return proxy.SidecarScope.destinationRulesByNames[ConfigKey{
+		Kind:      gvk.DestinationRule,
```

Comment:
nit: kind is redundant

Reply:
agree, make it take a NamespacedName type. There are some defined, not sure if in `model` or if we need to just make a new one (doesn't even need to be externally visible).